Unable to build socket can with frdm_k64f

**Describe the bug**
Unable to compile socket can application with frdm_k64f board

**To Reproduce**
Steps to reproduce the behavior:
1. cd samples/net/sockets/can
2. west build -b frdm_k64f
3. See error

**Expected behavior**
The socket can sample should build without any error.

**Impact**
socket can sample build with errors

**Logs**
````
/home/navin/workspace/zephyrproject/zephyr/drivers/can/can_mcux_flexcan.c:815:1: warning: extra tokens at end of #include directive
  815 | #define FLEXCAN_DEVICE_SOCKET_CAN(id)     \
      | ^
In file included from ../../../../../include/sys/util_macro.h:34,
                 from ../../../../../include/sys/util.h:17,
                 from ../../../../../include/sched_priq.h:9,
                 from ../../../../../include/kernel_includes.h:23,
                 from ../../../../../include/kernel.h:17,
                 from ../../../../../include/zephyr.h:18,
                 from /home/navin/workspace/zephyrproject/zephyr/drivers/can/can_mcux_flexcan.c:9:
zephyr/include/generated/devicetree_unfixed.h:5988:57: error: expected declaration specifiers or '...' before numeric constant
 5988 | #define DT_FOREACH_OKAY_INST_nxp_kinetis_flexcan(fn) fn(0)
      |                                                         ^
../../../../../include/sys/util_internal.h:65:26: note: in definition of macro '__DEBRACKET'
   65 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
../../../../../include/sys/util_internal.h:57:2: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   57 |  __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |  ^~~~~~~~~~~~~~~~~~~~
../../../../../include/sys/util_internal.h:52:2: note: in expansion of macro '__COND_CODE'
   52 |  __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~
../../../../../include/sys/util_macro.h:117:2: note: in expansion of macro 'Z_COND_CODE_1'
  117 |  Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |  ^~~~~~~~~~~~~
../../../../../include/devicetree.h:2074:2: note: in expansion of macro 'COND_CODE_1'
 2074 |  COND_CODE_1(DT_HAS_COMPAT_STATUS_OKAY(DT_DRV_COMPAT), \
      |  ^~~~~~~~~~~
../../../../../include/sys/util_internal.h:106:36: note: in expansion of macro 'DT_FOREACH_OKAY_INST_nxp_kinetis_flexcan'
  106 | #define UTIL_PRIMITIVE_CAT(a, ...) a##__VA_ARGS__
      |                                    ^
../../../../../include/sys/util_internal.h:105:26: note: in expansion of macro 'UTIL_PRIMITIVE_CAT'
  105 | #define UTIL_CAT(a, ...) UTIL_PRIMITIVE_CAT(a, __VA_ARGS__)
      |                          ^~~~~~~~~~~~~~~~~~
../../../../../include/devicetree.h:2075:8: note: in expansion of macro 'UTIL_CAT'
 2075 |       (UTIL_CAT(DT_FOREACH_OKAY_INST_,  \
      |        ^~~~~~~~
/home/navin/workspace/zephyrproject/zephyr/drivers/can/can_mcux_flexcan.c:840:1: note: in expansion of macro 'DT_INST_FOREACH_STATUS_OKAY'
  840 | DT_INST_FOREACH_STATUS_OKAY(FLEXCAN_DEVICE_SOCKET_CAN)
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/navin/workspace/zephyrproject/zephyr/drivers/can/can_mcux_flexcan.c:839:
/home/navin/workspace/zephyrproject/zephyr/drivers/can/socket_can_generic.h:120:34: warning: 'socket_can_context_1' defined but not used [-Wunused-variable]
  120 | static struct socket_can_context socket_can_context_1;
      |                                  ^~~~~~~~~~~~~~~~~~~~
/home/navin/workspace/zephyrproject/zephyr/drivers/can/socket_can_generic.h:113:26: warning: 'socket_can_api' defined but not used [-Wunused-variable]
  113 | static struct canbus_api socket_can_api = {
[130/188] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/net_context.c.obj
ninja: build stopped: subcommand failed.
````

**Environment (please complete the following information):**
  - Commit SHA: af67564573f5d7a76c703e7af57f5660c176679d

--- a/drivers/can/can_loopback.c
+++ b/drivers/can/can_loopback.c
@@ -303,6 +303,8 @@ DEVICE_DEFINE(can_loopback_1, CONFIG_CAN_LOOPBACK_DEV_NAME,
 
 #include "socket_can_generic.h"
 
+static struct socket_can_context socket_can_context_1;
+
 static int socket_can_init_1(const struct device *dev)
 {
 	const struct device *can_dev = DEVICE_GET(can_loopback_1);

--- a/drivers/can/can_mcp2515.c
+++ b/drivers/can/can_mcp2515.c
@@ -977,6 +977,8 @@ DEVICE_DT_INST_DEFINE(0, &mcp2515_init, device_pm_control_nop,
 
 #include "socket_can_generic.h"
 
+static struct socket_can_context socket_can_context_1;
+
 static int socket_can_init(const struct device *dev)
 {
 	const struct device *can_dev = DEVICE_DT_INST_GET(1);

--- a/drivers/can/can_mcux_flexcan.c
+++ b/drivers/can/can_mcux_flexcan.c
@@ -820,11 +820,12 @@ static const struct can_driver_api mcux_flexcan_driver_api = {
 DT_INST_FOREACH_STATUS_OKAY(FLEXCAN_DEVICE_INIT_MCUX)
 
 #if defined(CONFIG_NET_SOCKETS_CAN)
-#include "socket_can_generic.h"						\
+#include "socket_can_generic.h"
 #define FLEXCAN_DEVICE_SOCKET_CAN(id)					\
+	static struct socket_can_context socket_can_context_##id;	\
 	static int socket_can_init_##id(const struct device *dev)	\
 	{								\
-		struct device *can_dev = DEVICE_DT_INST_GET(id);	\
+		const struct device *can_dev = DEVICE_DT_INST_GET(id);	\
 		struct socket_can_context *socket_context = dev->data;	\
 		LOG_DBG("Init socket CAN device %p (%s) for dev %p (%s)", \
 			dev, dev->name, can_dev, can_dev->name);	\

--- a/drivers/can/can_stm32.c
+++ b/drivers/can/can_stm32.c
@@ -1190,6 +1190,8 @@ static void config_can_1_irq(CAN_TypeDef *can)
 
 #include "socket_can_generic.h"
 
+static struct socket_can_context socket_can_context_1;
+
 static int socket_can_init_1(const struct device *dev)
 {
 	const struct device *can_dev = DEVICE_DT_GET(DT_NODELABEL(can1));
@@ -1279,6 +1281,8 @@ static void config_can_2_irq(CAN_TypeDef *can)
 #if defined(CONFIG_NET_SOCKETS_CAN)
 
 #include "socket_can_generic.h"
+
+static struct socket_can_context socket_can_context_2;
 
 static int socket_can_init_2(const struct device *dev)
 {

--- a/drivers/can/socket_can_generic.h
+++ b/drivers/can/socket_can_generic.h
@@ -13,7 +13,9 @@
 #ifndef ZEPHYR_DRIVERS_CAN_SOCKET_CAN_GENERIC_H_
 #define ZEPHYR_DRIVERS_CAN_SOCKET_CAN_GENERIC_H_
 
+#define SOCKET_CAN_NAME_0 "SOCKET_CAN_0"
 #define SOCKET_CAN_NAME_1 "SOCKET_CAN_1"
+#define SOCKET_CAN_NAME_2 "SOCKET_CAN_2"
 #define SEND_TIMEOUT K_MSEC(100)
 #define RX_THREAD_STACK_SIZE 512
 #define RX_THREAD_PRIORITY 2
@@ -116,8 +118,6 @@ static struct canbus_api socket_can_api = {
 	.close = socket_can_close,
 	.setsockopt = socket_can_setsockopt,
 };
-
-static struct socket_can_context socket_can_context_1;
 
 static inline void rx_thread(void *ctx, void *unused1, void *unused2)
 {

--- a/samples/net/sockets/can/sample.yaml
+++ b/samples/net/sockets/can/sample.yaml
@@ -2,7 +2,6 @@ common:
   tags: net socket can CAN
   depends_on: can
   harness: can
-  platform_allow: stm32f072b_disco nucleo_l432kc
 sample:
   description: Test BSD sockets CAN support
   name: Socket CAN example

